xilem_core/src/views/memoize.rs

```diff
@@ -1,65 +1,76 @@
 // Copyright 2024 the Xilem Authors
 // SPDX-License-Identifier: Apache-2.0
 
+use core::marker::PhantomData;
+use core::mem::size_of;
+
 use crate::{MessageResult, Mut, View, ViewId, ViewPathTracker};
 
 /// A view which supports Memoization.
 ///
 /// The story of Memoization in Xilem is still being worked out,
 /// so the details of this view might change.
-pub struct Memoize<D, F> {
-    data: D,
-    child_cb: F,
-}
-
-pub struct MemoizeState<State, Action, Context, Message, V>
-where
-    Context: ViewPathTracker,
-    V: View<State, Action, Context, Message>,
-{
-    view: V,
-    view_state: V::ViewState,
-    dirty: bool,
+pub struct Memoize<Data, InitView, State, Action> {
+    data: Data,
+    init_view: InitView,
+    phantom: PhantomData<fn() -> (State, Action)>,
 }
 
-impl<D, V, F> Memoize<D, F>
-where
-    F: Fn(&D) -> V,
-{
-    const ASSERT_CONTEXTLESS_FN: () = {
-        assert!(
-            core::mem::size_of::<F>() == 0,
-            "
+macro_rules! non_capturing_closure_string {
```

Comment:
We need a comment to explain why we're writing that this way. I'd expect `assert!(..., "{}", ASSERTION_STRING)` to be the idiomatic Rust here, where the value is stored in a constant.

Did that run into the rustfmt issue?

Reply:
Yeah thinking about it, that's probably better, no I didn't run into it, just haven't thought about it.